KIPs/kip-draft_kgp6.md

```diff
@@ -0,0 +1,672 @@
+     * @return receiver is the address of the receiver
+     * @return amount is the fund allocated to the receiver
+
+     */
+    function getReceiver(address _receiverAddress)
+        public
+        view
+        returns (address, uint256)
+    {
+        require(receiverExists(_receiverAddress), "Receiver does not exist");
+        uint256 index = getReceiverIndex(_receiverAddress);
+        Receiver memory receiver = receivers[index];
+        return (receiver.receiver, receiver.amount);
+    }
+
+    /**
+     * @dev check whether _receiverAddress is registered
+     * @param _receiverAddress is the address of the receiver
+     */
+    function receiverExists(address _receiverAddress)
+        public
+        view
+        returns (bool)
+    {
+        require(_receiverAddress != address(0), "Invalid address");
+        for (uint8 i = 0; i < receivers.length; i++) {
+            if (receivers[i].receiver == _receiverAddress) {
+                return true;
+            }
+        }
+    }
+
+    /**
+     * @dev get index of the receiver in the receivers array
+     * @param _receiverAddress is the address of the receiver
+     */
+    function getReceiverIndex(address _receiverAddress)
+        public
+        view
+        returns (uint256)
+    {
+        for (uint256 i = 0; i < receivers.length; i++) {
+            if (receivers[i].receiver == _receiverAddress) {
+                return i;
+            }
+        }
+        revert("Receiver does not exist");
+    }
+
+    /**
+     * @dev to calculate the sum of receiver funds
+     * @return treasuryAmount the sum of funds allocated to receivers
+     */
+    function getTreasuryAmount() public view returns (uint256 treasuryAmount) {
+        for (uint8 i = 0; i < receivers.length; i++) {
+            treasuryAmount += receivers[i].amount;
+        }
+        return treasuryAmount;
+    }
+
+    /**
+     * @dev gets the length of senders list
+     */
+    function getSenderCount() public view returns (uint256) {
+        return senders.length;
+    }
+
+    /**
+     * @dev gets the length of receivers list
+     */
+    function getReceiverCount() public view returns (uint256) {
+        return receivers.length;
+    }
+
+    /**
+     * @dev allback function to revert any payments
+     */
+    fallback() external payable {
+        revert("This contract does not accept any payments");
+    }
+
+    /**
+     * @dev Helper function to check the address is contract addr or EOA
+     */
+    function isContractAddr(address _addr) public view returns (bool) {
+        uint256 size;
+        assembly {
+            size := extcodesize(_addr)
+        }
+        return size > 0;
+    }
+}
+```
+
+## Test Cases
+
+<!--Test cases for an implementation are mandatory for KIPs that are affecting consensus changes. Other KIPs can choose to include links to test cases if applicable.-->
+
+https://github.com/Krustuniverse-Klaytn-Group/treasury-rebalance/blob/dev/test/TreasuryRebalance.js
```

Comment:
Will this repo be open to public?

Reply:
no this wont be open to public. I will attach the file here instead of link

Reply:
If this will not be public, there is no reason to add this here. Since this KIPs repo is a public repo. A public repo with a private link?

Reply:
made the repo to public and changed the link